api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -0,0 +1,172 @@
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		Condition condition = conditionService.getConditionByUuid(uuid);
+		Assert.assertEquals(uuid, condition.getUuid());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
+		Assert.assertNull(conditionService.getConditionByUuid("invalid uuid"));
+	}
+
+	/**
+	 * @see ConditionService#getCondition(Integer) 
+	 */
+	@Test
+	public void getCondition_shouldFindConditionGivenValidId() {
+		Condition condition = conditionService.getCondition(1);
+		Assert.assertEquals(ConditionClinicalStatus.INACTIVE, condition.getClinicalStatus());
+		Assert.assertEquals(ConditionVerificationStatus.PROVISIONAL, condition.getVerificationStatus());
+		Assert.assertEquals("2cc6880e-2c46-15e4-9038-a6c5e4d22fb7", condition.getUuid());
+	}
+
+	/**
+	 * @see ConditionService#getActiveConditions(Patient)
+	 */
+	@Test
+	public void getActiveConditions_shouldGetActiveConditions() {
+		List<Condition> activeConditions = conditionService.getActiveConditions(patientService.getPatient(2));
+		Condition firstActiveCondition = activeConditions.get(0);
+		Condition secondActiveCondition = activeConditions.get(1);
+		String firstUUID = "2cc6880e-2c46-11e4-9138-a6c5e4d20fb7";
+		String secondUUID = "2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		Assert.assertTrue(activeConditions.size() == 2);
+		Assert.assertEquals(firstUUID,firstActiveCondition.getUuid());
+		Assert.assertEquals(secondUUID, secondActiveCondition.getUuid());
+	}
+	
+	
+
+	/**
+	 * @see ConditionService#voidCondition(Condition, String) 
+	 */
+	@Test
+	public void voidCondition_shouldVoidConditionSuccessfully(){
+		String voidReason = "Test Reason";
+		Condition nonVoidedCondition = conditionService.getCondition(2);
+		Assert.assertFalse(nonVoidedCondition.getVoided());
+		Assert.assertNull(nonVoidedCondition.getVoidReason());
+		Assert.assertNull(nonVoidedCondition.getDateVoided());
+		Assert.assertNull(nonVoidedCondition.getVoidedBy());
+		
+		Condition voidedCondition = conditionService.voidCondition(nonVoidedCondition, voidReason);
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, voidedCondition.getVerificationStatus());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, voidedCondition.getClinicalStatus());
+		Assert.assertEquals("2cc6880e-2c46-11e4-9138-a6c5e4d20fb7", voidedCondition.getUuid());
+		Assert.assertTrue(voidedCondition.getVoided());
+		Assert.assertEquals(voidReason, voidedCondition.getVoidReason());
+		Assert.assertNotNull(voidedCondition.getDateVoided());
+		Assert.assertEquals(Context.getAuthenticatedUser(), voidedCondition.getVoidedBy());
+	}
+
+	/**
+	 * @see ConditionService#unvoidCondition(Condition) 
+	 */
+	@Test
+	public void unvoidCondition_shouldUnvoidConditionSuccessfully(){
+		Condition voidedCondition = conditionService.voidCondition(conditionService.getCondition(2), "Test Reason");
```

Comment:
Instead of voiding a condition here, just load one from the database which is already voided.

Reply:
Okay. Done